cob_teleop/ros/src/cob_teleop.cpp

```diff
@@ -435,7 +436,21 @@ void CobTeleop::joy_cb(const sensor_msgs::Joy::ConstPtr &joy_msg)
     safe_mode_ = true;
   }
 
-  if(run_button_>=0 && run_button_<(int)joy_msg->buttons.size() && joy_msg->buttons[run_button_]==1)
+  if(joy_config_switch_.compare("X") == 0 && axis_run_ < joy_msg->axes.size())
+  {
+    if(!axis_run_updated_)
+    {
+      if(joy_msg->axes[axis_run_] != 0)
+        axis_run_updated_ = true;
+      else
+        run_factor_ = 1.0;
+    }
+    if(axis_run_updated_)
+    {
+      run_factor_ = 1 + ((run_factor_param_ - 1.0) / -2.0) * (joy_msg->axes[axis_run_] - 1.0);
+    }
+  }
```

Comment:
could this not be simplified to:
```suggestion
  if(joy_config_switch_.compare("X") == 0 && axis_run_ < joy_msg->axes.size())
  {
    if(joy_msg->axes[axis_run_] != 0)
    {
      run_factor_ = 1 + ((run_factor_param_ - 1.0) / -2.0) * (joy_msg->axes[axis_run_] - 1.0);
    }
    else
    {
      run_factor_ = 1.0;
    }
  }
```

then you don't need the `axis_run_updated_` variable....

Reply:
The joy driver works event based. That means that the values in the joy message are only updated when a button/axis on the joystick is pressed. The real range of the trigger buttons is from 1 (not pressed) to -1 (fully pressed). That means that the values for the trigger axis will stay 0 as long as no event for them was fired but cob_teleop expects a value of 1 when the trigger axis are not pressed. Thats why we need the `axis_run_updated_`

Reply:
ok, clear now....